### ISSUE TITLE
Example of using cxx in an external mixed bazel C++ & Rust environment?

### ISSUE THREAD

**Issue by sthornington** [dtolnay/cxx#1365]
Bazel is not my first language, but I am having substantial difficulty trying to import cxx into a bazel environment, both using bzlmod/MODULE and also using WORKSPACE.  

Is there an example out there of a simple project with trivial actual cxx bridging but demonstrating the bazel details for bringing it down, building it, and generating the stubs for each side?  Thanks (in advance).

**Comment by AM5800** [comment]
@sthornington sorry for reopening an old thread, but were you eventually able to find such example?

**Comment by sthornington** [comment]
No not really, but I did eventually get it working.  My bazel is a lot better than it was then, and LLMs are very good at it. The nested repo with the crate_universe declared in the embedded build file string, within my WORKSPACE, was important, as was figuring out the specific rule/macro to invoke the binary.

I found it quite complicated to ultimately make the C++ library that contained the bridge module, but eventually figured that out too. Wound up being ~six bazel targets for the invocation, headers, C++ library, rust static library, rust library etc to get to the point where one could have just one or two additional deps to link a C++ or a Rust target against the bridge.

A colleague had an even more complicated situation trying to package up a bazel C++ library into a rust crate, he wound up compiling an archive to run a script to deduce all the objects that went into it to re-archive it with additional dependencies to make a super-archive.

All this to say, we eventually got it all working, but more bazel examples might have shortened our journey.
